packages/wrangler/src/core/register-yargs-command.ts

```diff
@@ -157,21 +159,47 @@ function createHandler(def: CommandDefinition) {
 						MIXED_MODE: args.experimentalMixedMode ?? false,
 					};
 
-			await run(experimentalFlags, () =>
-				def.handler(args, {
-					config:
-						def.behaviour?.provideConfig ?? true
-							? readConfig(args, {
-									hideWarnings: !(def.behaviour?.printConfigWarnings ?? true),
-									useRedirectIfAvailable:
-										def.behaviour?.useConfigRedirectIfAvailable,
-								})
-							: defaultWranglerConfig,
+			await run(experimentalFlags, () => {
+				const config =
+					def.behaviour?.provideConfig ?? true
+						? readConfig(args, {
+								hideWarnings: !(def.behaviour?.printConfigWarnings ?? true),
+								useRedirectIfAvailable:
+									def.behaviour?.useConfigRedirectIfAvailable,
+							})
+						: defaultWranglerConfig;
+
+				if (def.behaviour?.warnIfMultipleEnvsConfiguredButNoneSpecified) {
+					if (!args.env) {
+						const { configPath } = findWranglerConfig(process.cwd(), {
+							useRedirectIfAvailable: true,
+						});
```

Comment:
Can you use `userConfigPath` available on the `config` object?

Reply:
mh... I don't think so 🤔 

`userConfigPath` is the non-redirected config path right? so a redirected configuration would not be taken into account, right?

What I'm thinking of is a scenario in which a user has different environments in their config file, then they use some tool (e.g. nitro, vite) to build their worker targeting a specific environment (which builds the appropriate environment-less redirected config), afterwords they run `wrangler deploy` for that command.
In such case we don't want `wrangler deploy` to be called with an environment (since that's not supported in redirected configs) so not taking into consideration the (environment-less) redirected environment and present the warning would be incorrect, wouldn't it?

Reply:
Good point. I didn't think of that scenario. Indeed we should not warn if the actual config being used is a redirected one with only a top level environment. In which case I alter my suggestion, which is to use the `config.configPath` property from the read config, rather than trying to find the config a second time.